### ISSUE TITLE
KVM hosts unable to introspect newer Windows builds

### ISSUE THREAD

**Issue by namidairo** [libvmi/libvmi#598]
Due to the lack of access to the LSTAR/GSTAR model specific registers on KVM hosts, the init_from_rekall_profile changes from #564 will fail out on systems with no KiInitialPCR but possess KiSystemCall64Shadow AND KiDivideErrorFault. (Ie. The newer builds with the Spectre mitigations)

I have verified that the KiDivideErrorFault symbol at least still exists, but I haven't confirmed if that method still works.

A better solution however, might be to skip over to the next method rather than having every failure hit `goto done;`

**Comment by buekera** [comment]
This issue seems to be related to [this one](https://github.com/libvmi/libvmi/issues/604), isn't it?

**Comment by buekera** [comment]
@namidairo Did you make any progress on this issue?

Can you actually tell me if this is more an issue with KVM or Windows 10?

**Comment by namidairo** [comment]
(Finally) got around to looking at this properly, is actually quite a simple fix, but requires that we amend all the qemu-kvm patches.

```
diff --git a/target/i386/helper.c b/target/i386/helper.c
index e695f8ba7a..df83f6b897 100644
--- a/target/i386/helper.c
+++ b/target/i386/helper.c
@@ -506,6 +506,8 @@ void x86_cpu_dump_state(CPUState *cs, FILE *f, fprintf_function cpu_fprintf,
             cpu_fprintf(f, "DR%d=%016" PRIx64 " ", i, env->dr[i]);
         cpu_fprintf(f, "\nDR6=%016" PRIx64 " DR7=%016" PRIx64 "\n",
                     env->dr[6], env->dr[7]);
+        cpu_fprintf(f, "CSTAR=%016" PRIx64 " LSTAR=%016" PRIx64 "\n",
+                    env->cstar, env->lstar);
     } else
 #endif
     {

```
And then we can just simply parse the register like we do for the rest like so

```
diff --git a/libvmi/driver/kvm/kvm.c b/libvmi/driver/kvm/kvm.c
index ca11948..82db738 100644
--- a/libvmi/driver/kvm/kvm.c
+++ b/libvmi/driver/kvm/kvm.c
@@ -1849,6 +1849,12 @@ kvm_get_vcpureg(
         case MSR_EFER:
             *value = parse_reg_value("EFER", regs);
             break;
+        case MSR_CSTAR:
+            *value = parse_reg_value("CSTAR", regs);
+            break;
+        case MSR_LSTAR:
+            *value = parse_reg_value("LSTAR", regs);
+            break;
         default:
             if ( VMI_PM_IA32E == vmi->page_mode) {
                 switch (reg) {
```

**Comment by namidairo** [comment]
Solved by #763
